src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java

```diff
@@ -1,6 +1,45 @@
 package com.vinylteam.vinyl.service.impl;
 
+import com.vinylteam.vinyl.dao.jdbc.JdbcUserDao;
+import com.vinylteam.vinyl.entity.User;
 import com.vinylteam.vinyl.service.UserService;
+import org.postgresql.ds.PGSimpleDataSource;
 
 public class DefaultUserService implements UserService {
+
+    private final JdbcUserDao jdbcUserDao;
+
+    public DefaultUserService(PGSimpleDataSource dataSource) {
+        this.jdbcUserDao = new JdbcUserDao(dataSource);
+    }
+
+    @Override
+    public boolean add(User user) {
+        return jdbcUserDao.add(user);
+    }
+
+    @Override
+    public void updatePassword(String email, String newPassword, int iterations, String newSalt) {
+
+    }
+
+    @Override
+    public User getByEmail(String email) {
+        return jdbcUserDao.getByEmail(email);
+    }
+
+    @Override
+    public boolean removeUser(String email) {
+        return false;
+    }
+
+    @Override
+    public int countAll() {
```

Comment:
what do this method do? And why do we need it?

Reply:
CountAll helps test DAO methods like "getByEmail", "add", and "remove" in the future - it allows to check that amount of users in db didn't change after getByEmail, didn't change after adding user with email that exists in DB, that we add exactly one user and so on.

I need it for testing.

Reply:
> CountAll helps test DAO methods like "getByEmail", "add", and "remove" in the future - it allows to check that amount of users in db didn't change after getByEmail, didn't change after adding user with email that exists in DB, that we add exactly one user and so on.
> 
> I need it for testing.

If it is using only for testing please change method access modifier to package-private(default).

Reply:
ok

Reply:
@AlexKievDeveloper when I make it package-private 'countAll()' in 'com.vinylteam.vinyl.dao.jdbc.JdbcUserDao' clashes with 'countAll()' in 'com.vinylteam.vinyl.dao.UserDao'; attempting to assign weaker access privileges ('package-private'); was 'public'

I can't make interface package-private.
Should I delete this method from interface and leave only in implementation, or should I leave it public?